tests/test_tagging.py

```diff
@@ -0,0 +1,301 @@
+        ingesterConfig["forwarderStagingDirectory"] = self.forwarderStagingDir
+
+        self.badDir = tempfile.mkdtemp()
+        butlerConfig = ingesterConfig["butlers"][0]["butler"]
+        butlerConfig["badFileDirectory"] = self.badDir
+        self.stagingDirectory = tempfile.mkdtemp()
+        butlerConfig["stagingDirectory"] = self.stagingDirectory
+
+        self.repoDir = tempfile.mkdtemp()
+        butlerConfig["repoDirectory"] = self.repoDir
+
+        self.collections = butlerConfig["collections"]
+        print(f'self.collections = {self.collections}')
+
+        # copy the FITS file to it's test location
+
+        subDir = tempfile.mkdtemp(dir=self.forwarderStagingDir)
+        self.destFile = os.path.join(subDir, fits_name)
+        copyfile(fitsFile, self.destFile)
+
+        # setup directory to scan for files in the forwarder staging directory
+        # and ensure one file is there
+        ingesterConfig = config["ingester"]
+        forwarder_staging_dir = ingesterConfig["forwarderStagingDirectory"]
+        scanner = DirectoryScanner([forwarder_staging_dir])
+        files = scanner.getAllFiles()
+        self.assertEqual(len(files), 1)
+
+        # create the file ingester, get all tasks associated with it, and
+        # create the tasks
+        ingester = FileIngester(ingesterConfig)
+        butler_tasks = ingester.getButlerTasks()
+
+        task_list = []
+        for butler_task in butler_tasks:
+            task = asyncio.create_task(butler_task())
+            task_list.append(task)
+
+        # check to see that the file is there before ingestion
+        self.assertTrue(os.path.exists(self.destFile))
+
+        # trigger the ingester by sending it a "message"
+        msg = {}
+        msg['CAMERA'] = "COMCAM"
+        msg['OBSID'] = "CC_C_20190530_000001"
+        msg['FILENAME'] = self.destFile
+        msg['ARCHIVER'] = "CCArchiver"
+        await ingester.ingest(msg)
+
+        # make sure staging area is now empty
+        files = scanner.getAllFiles()
+        self.assertEqual(len(files), 0)
+
+        # Check to see that the file was ingested.
+        # Recall that files start in teh forwarder staging area, and are
+        # moved to the OODS staging area before ingestion. On "direct"
+        # ingestion, this is where the file is located.  This is a check
+        # to be sure that happened.
+        name = self.strip_prefix(self.destFile, self.forwarderStagingDir)
+        staged_file = os.path.join(self.stagingDirectory, name)
+        self.assertTrue(os.path.exists(staged_file))
+
+        # this file should now not exist
+        self.assertFalse(os.path.exists(self.destFile))
+
+        # add one more task, whose sole purpose is to interrupt the others by
+        # throwing an exception. This is used to exit all tasks.
+        task_list.append(asyncio.create_task(self.interrupt_me()))
+
+        return staged_file, task_list
+
+    async def testTaggedFileTestCase(self):
```

Comment:
Can we add a docstring briefly explaining the purpose of this function?

Reply:
It was covered in the extensive comment at the beginning of the class, but I added a brief summary here as well, as requested.